### PR TITLE
Fetch Product By ID

### PR DESCRIPTION
**Pull request by JoshuaJohnson8848**
Fetch Product By ID Feature Added

### CODE REVIEW

File: controller/product.js [hunk 1]
```diff
@@ -38,12 +38,30 @@ exports.getAll = async (req, res, next) => {
...
+  } catch (err) {
+    if (!err.status) {
+      err.status = 500;
+    }
+    next(err);
+  }
+};
+
+exports.getById = async (req, res, next) => {
+  const productId = req.params.productId;
```
Comment (freduz) [inline controller/product.js]: use optional chaining there like "req.params?.productId"

File: controller/product.js [hunk 2]
```diff
@@ -38,12 +38,30 @@ exports.getAll = async (req, res, next) => {
...
+      error.status = 422;
+      throw error;
+    }
     res
       .status(200)
-      .json({
-        message: 'All Products Fetched Successfully',
-        product: products,
-      });
+      .json({ message: 'Product Fetched Successfully', product: product });
```
Comment (freduz) [inline controller/product.js]: dont want to write product:product use can simply use json({message:'message goes here',product})

File: controller/product.js (hunk 1 above)
Comment (JoshuaJohnson8848) [reply controller/product.js]: @freduz changed

File: controller/product.js (hunk 2 above)
Comment (JoshuaJohnson8848) [reply controller/product.js]: @freduz solved

**Pull request merged**